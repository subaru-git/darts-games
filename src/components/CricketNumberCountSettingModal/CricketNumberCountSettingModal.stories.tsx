/* eslint-disable import/named */
import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import CricketNumberCountSettingModal from './CricketNumberCountSettingModal'

export default {
  title: 'Components/CricketNumberCountSettingModal',
  component: CricketNumberCountSettingModal,
} as ComponentMeta<typeof CricketNumberCountSettingModal>

const Template: ComponentStory<typeof CricketNumberCountSettingModal> = (args) => <CricketNumberCountSettingModal {...args} />

export const Default = Template.bind({})
Default.args = {}
