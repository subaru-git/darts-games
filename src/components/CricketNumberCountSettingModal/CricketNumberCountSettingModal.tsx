import React, { FC } from 'react'
import {
  Box,
  Button,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  NumberDecrementStepper,
  NumberIncrementStepper,
  NumberInput,
  NumberInputField,
  NumberInputStepper,
  Text,
} from '@chakra-ui/react'
import CricketNumberCountDescription from '../CricketNumberCountDescription'

type CricketNumberCountSettingModalProps = {
  isOpen: boolean
  onClose: () => void
  targetCount: number
  onNewGame: (targetCount: number) => void
}

const CricketNumberCountSettingModal: FC<CricketNumberCountSettingModalProps> = ({
  isOpen,
  onClose,
  targetCount,
  onNewGame,
}) => {
  const [innerTargetCount, setInnerTargetCount] = React.useState(targetCount)
  return (
    <Modal isOpen={isOpen} onClose={onClose} size='2xl'>
      <ModalOverlay />
      <ModalContent>
        <ModalHeader>Cricket Number Count Settings</ModalHeader>
        <ModalCloseButton />
        <ModalBody>
          <Box py={4}>
            <CricketNumberCountDescription />
          </Box>
          <Text fontSize='sm'>Target Count</Text>
          <NumberInput
            defaultValue={innerTargetCount}
            onChange={(value) => setInnerTargetCount(parseInt(value))}
          >
            <NumberInputField />
            <NumberInputStepper>
              <NumberIncrementStepper />
              <NumberDecrementStepper />
            </NumberInputStepper>
          </NumberInput>
        </ModalBody>
        <ModalFooter>
          <Button
            colorScheme='blue'
            mr={3}
            onClick={() => {
              onNewGame(innerTargetCount)
            }}
          >
            New Game
          </Button>
          <Button variant='ghost' onClick={onClose}>
            Cancel
          </Button>
        </ModalFooter>
      </ModalContent>
    </Modal>
  )
}

export default CricketNumberCountSettingModal
