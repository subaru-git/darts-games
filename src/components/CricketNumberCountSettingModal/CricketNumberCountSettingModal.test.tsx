import { render } from '@testing-library/react'
import CricketNumberCountSettingModal from './CricketNumberCountSettingModal'

test('should rendering', () => {
  const { container } = render(<CricketNumberCountSettingModal />)
  expect(container).toMatchSnapshot()
})
