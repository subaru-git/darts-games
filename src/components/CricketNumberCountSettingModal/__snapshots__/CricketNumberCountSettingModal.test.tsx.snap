// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `<div />`;
